lib/config.js

```diff
@@ -14,24 +22,93 @@ function getNodeVersion () {
   return [Number(version[1]), Number(version[2])];
 }
 
+function updateBuildInfo () {
+  return new B((resolve) => {
+    getGitRev().then((sha) => { // eslint-disable-line promise/prefer-await-to-then
+      if (_.isEmpty(sha)) {
+        logger.warn(`Cannot retrieve git revision for Appium version ${APPIUM_VER}`);
+        return resolve();
+      }
+
+      BUILD_INFO['git-sha'] = sha;
+      getGitTimestamp(sha)
+        .then((built) => { // eslint-disable-line promise/prefer-await-to-then
+          if (_.isEmpty(built)) {
+            logger.warn(`Cannot retrieve the timestamp for Appium git commit ${sha}`);
+          } else {
+            BUILD_INFO.built = built;
+          }
+          resolve();
+        }).catch((e) => {
+          logger.warn(`Cannot retrieve the timestamp for Appium git commit ${sha}. ` +
+            `Original error: ${e.message}`);
+          resolve();
+        });
+    }).catch((e) => {
+      logger.warn(`Cannot retrieve git revision for Appium version ${APPIUM_VER}. ` +
+        `Original error: ${e.message}`);
+      resolve();
+    });
+  });
+}
+
 async function getGitRev () {
-  let rev = null;
-  try {
-    let {stdout} = await exec("git", ["rev-parse", "HEAD"], {rootDir});
-    rev = stdout.trim();
-  } catch (ign) {}
-  return rev;
+  if (await fs.exists(path.resolve(rootDir, GIT_META_ROOT))) {
+    const {stdout} = await exec(GIT_BINARY, ['rev-parse', 'HEAD'], {
+      cwd: rootDir
+    });
+    return stdout.trim();
+  } else {
+    const response = await request.get(`${GITHUB_API}/tags`, {
```

Comment:
Do we really want to start introducing the overhead of network calls just for the sake of getting the git SHA?  This means for each time the server starts it needs to make two network calls just to print out some information?

Reply:
Is it a problem? the calls are not blocking so we don't really care if and when they succeed. Although, this might give us (and clients) more useful information

Reply:
`await request.get` should be blocking.

For any released version of Appium, the exact SHA, as well as the tree at that point, can be retrieved through https://github.com/appium/appium/releases, or just by going to `https://github.com/appium/appium/tree/v<WHATEVER-VERSION>`.

Reply:
Is there any reason why it should be blocking? The idea is to never block `getStatus` responses and only include the retrieved information there when it is available for us.

> For any released version of Appium, the exact SHA, as well as the tree at that point, can be retrieved through https://github.com/appium/appium/releases, or just by going to https://github.com/appium/appium/tree/v<WHATEVER-VERSION>.

I do use requests to official API endpoints, which return the information as nicely formatted json.

Reply:
I just mean it can easily be retrieved manually. If you know the SHA you have about as many steps to get to the tree for that commit as if you just know the npm version.

For the blocking part, `await` ensures that. You could do something like
```js
request.get(`${GITHUB_API}/tags`, {
  headers: {
    'User-Agent': `Appium ${APPIUM_VER}`
  }
}).then(function (res) {
  const resBodyObj = util.safeJsonParse(res);
  if (_.isArray(resBodyObj)) {
    for (const {name, commit} of resBodyObj) {
      if (name === `v${APPIUM_VER}` && commit && commit.sha) {
        BUILD_INFO['git-sha'] = commit.sha;
        break;
      }
    }
  }
});
```
and not care that when the first call is finished this will not have anything in it.

Reply:
ok I got your point